packages/data-context/src/actions/FileActions.ts

```diff
@@ -35,4 +37,28 @@ export class FileActions {
 
     return await this.ctx.fs.stat(filePath)
   }
+
+  openFile (absolute: string, line: number = 1, column: number = 1) {
+    const binary = this.ctx.coreData.localSettings.preferences.preferredEditorBinary
+
+    if (!binary || !absolute) {
+      this.ctx.debug('cannot open file without binary')
+
+      return
+    }
+
+    if (binary === 'computer') {
+      try {
+        this.ctx.electronApi.showItemInFolder(absolute)
+      } catch (err) {
+        this.ctx.debug('error opening file: %s', (err as Error).stack)
+      }
+
+      return
+    }
+
+    launchEditor(`${absolute}:${line}:${column}`, `"${binary}"`, (__: unknown, errMsg: string) => {
+      this.ctx.debug('error opening file: %s', errMsg)
+    })
```

Comment:
Do you think this implementation should be done here or could it be done in another file?

Reply:
I guess, why? Where?